pkg/framework/plugins/removepodsviolatingnodetaints/node_taint.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package removepodsviolatingnodetaints
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/descheduler/pkg/apis/componentconfig"
+	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+	"sigs.k8s.io/descheduler/pkg/framework"
+	"sigs.k8s.io/descheduler/pkg/utils"
+)
+
+const PluginName = "RemovePodsViolatingNodeTaints"
+
+// RemovePodsViolatingNodeTaints evicts pods on the node which violate NoSchedule Taints on nodes
+type RemovePodsViolatingNodeTaints struct {
+	handle         framework.Handle
+	args           *componentconfig.RemovePodsViolatingNodeTaintsArgs
+	taintFilterFnc func(taint *v1.Taint) bool
+	podFilter      podutil.FilterFunc
+}
+
+var _ framework.Plugin = &RemovePodsViolatingNodeTaints{}
+var _ framework.DeschedulePlugin = &RemovePodsViolatingNodeTaints{}
+
+// New builds plugin from its arguments while passing a handle
+func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error) {
```

Comment:
should we just return *RemovePodsViolatingNodeTaints, casting it back to it in the test doesn't seem necessary?

Reply:
Once all strategies are migrated, each plugin will get stored under a [registry map](https://github.com/kubernetes-sigs/descheduler/pull/781/files#diff-a6d41c6c3467d9c9559603be28b857457382a9ad6d88e87419533821a731e587):
```go
type PluginBuilder = func(args runtime.Object, handle framework.Handle) (framework.Plugin, error)

type Registry = map[string]PluginBuilder

func NewRegistry() Registry {
	return Registry{
		nodeutilization.HighNodeUtilizationPluginName:          nodeutilization.NewHighNodeUtilization,
		nodeutilization.LowNodeUtilizationPluginName:           nodeutilization.NewLowNodeUtilization,
		podlifetime.PluginName:                                 podlifetime.New,
		removeduplicatepods.PluginName:                         removeduplicatepods.New,
		removefailedpods.PluginName:                            removefailedpods.New,
		removepodshavingtoomanyrestarts.PluginName:             removepodshavingtoomanyrestarts.New,
		removepodsviolatinginterpodantiaffinity.PluginName:     removepodsviolatinginterpodantiaffinity.New,
		removepodsviolatingnodeaffinity.PluginName:             removepodsviolatingnodeaffinity.New,
		removepodsviolatingnodetaints.PluginName:               removepodsviolatingnodetaints.New,
		removepodsviolatingtopologyspreadconstraint.PluginName: removepodsviolatingtopologyspreadconstraint.New,
		defaultevictor.PluginName:                              defaultevictor.New,
	}
}
```

Thus the reason why all plugins are expected to return the same interface.

Based on https://go.dev/play/p/JQllrX7szge go compiler does not allow assigning a function implementation where the signatures differ.

Reply:
ahh makes sense, the `PluginBuilder` type is the issue.